### DISCUSSION TITLE
Running ComfyUI on Intel NPUs with OpenVINO?

### DISCUSSION THREAD

**Discussion by Stanley5249** [comfyanonymous/ComfyUI#4987] [Q&A]
Hi everyone,

I've been searching for ways to run ComfyUI on Intel NPUs but haven't found much information. However, I did come across [OpenVINO™ AI Plugins for GIMP](https://github.com/intel/openvino-ai-plugins-gimp) which supports running Stable Diffusion on NPUs using OpenVINO. According to the OpenVINO documentation, you can compile a model for an NPU device with the following code:

```python
# compile the model for NPU device
core = ov.Core()
compiled_model = core.compile_model(ov_model, 'NPU')
```

I'm not very familiar with either ComfyUI or OpenVINO. Can anyone provide guidance or share their experiences with running ComfyUI on Intel NPUs? Any tips on integrating OpenVINO with ComfyUI would be greatly appreciated.

I also found another discussion mentioning NPUs, but it doesn't seem directly related: [ComfyUI for Intel Evo](https://github.com/comfyanonymous/ComfyUI/discussions/1942).

Thanks in advance for your help!

**Comment by simonlui** [comment]
It won't work since ComfyUI doesn't support an OpenVINO backend. I had not implemented it since it duplicated what IPEX already does for Intel Arc and would've been more complex at the time to bolt on. That being said, I may implement it in the future, but it requires some deeper plumbing work before something could be made to work.

**Comment by Iemand005** [comment]
I would really like this too, for older Intel iGPUs. OpenVINO seems to give reasonable performance on things like Iris Xe. I got IPEX working on Iris Xe even though it's supposedly not supported, but it's really messy and performance isn't great. I'm willing to dig into the python code for this if necessary.
